arrow/src/main/java/org/apache/iceberg/arrow/vectorized/parquet/VectorizedParquetDefinitionLevelReader.java

```diff
@@ -358,7 +358,8 @@ class FixedLengthDecimalReader extends BaseReader {
     protected void nextVal(
         FieldVector vector, int idx, ValuesAsBytesReader valuesReader, int typeWidth, byte[] byteArray) {
       valuesReader.getBuffer(typeWidth).get(byteArray, 0, typeWidth);
-      ((DecimalVector) vector).setBigEndian(idx, byteArray);
+      byte[] vectorBytes = DecimalVectorUtil.padBigEndianBytes(byteArray, DecimalVector.TYPE_WIDTH);
```

Comment:
This looks like a place where we could reuse a buffer rather than allocating in `padBigEndianBytes` every time.

Reply:
I did some testing, and reusing the buffer was a little bit slower, partly because we need to always to fill the buffer to zero out the last value.

Reply:
One thing that was a little bit faster was to bypass `DecimalVector.setBigEndian()`, convert to little endian (if needed) and copy the bytes directly to the value buffer.

Reply:
Also one thing to note is that the benchmark isn't quite right. Decimal(20,5) will end up taking 9 bytes and will thus use a fixed length byte array instead of long or int encoding. And fixed length byte arrays aren't dictionary encoded in Parquet v1. That explains why the decimal benchmark is much slower than the other data types (which are dictionary encoded).

Reply:
(It looks like dictionary encoding for fixed length byte arrays wouldn't work correctly anyway, I may follow up with a fix for that)

Reply:
Here's the [PR](https://github.com/apache/iceberg/pull/5198) that has a fix for the dictionary encoding